### ISSUE TITLE
Potential bug: units of different type stacking on top of each other

### ISSUE THREAD

**Issue by mongoose1130** [triplea-game/triplea#7053]
I've noticed since moving up to 2.0 (various releases to current) that sometimes units of differing types will overlap on the same placement box. I've seen it self-correct at different sequence points of the turn. This behavior is also inconsistent; meaning it can happen in territory A one time, but the next time it will be territory b and territory A will display fine.

I'd say this is low priority as it doesn't break anything - just a display glitch.

**Comment by Cernelius** [comment]
@mongoose1130 Take a look at your coordinates. I assume you have two or more of the same pair.

One of the issues of the placement picker is that it easily generates more than one pair of coordinates, if your finger is not very firm.

**Comment by mongoose1130** [comment]
@Cernelius that is what it was - thanks for the tip!

**Comment by Cernelius** [comment]
This is still a problem that I think deserves to be tracked.

Currently, the placement picker very easily creates two or more of the same coordinates if only you are a bit slow at clicking. What's worse, when multiple coordinates are thus created, it is impossible visually to detect them, as they will just appear as a single one, in the placement picker display.

This is an issue that it is a substantial problem for mapmaking, whose impact is limited mostly by the fact that I believe the placement picker is rarely used (probably due to its bad usability) (and this is not very good, either, as the automatic finder utility is very inefficient and wasteful).

Also the fact that I had to inform you of this issue, while, by now, you are a fairly experienced mapmaker, I think demostrates how this behaviour is hard even to make sense of.

**Comment by mongoose1130** [comment]
You are correct in that I would have not seen the placements had I not just went in to the tool and basically deleted all of the placements from the territory affected and I noticed I had to right-click twice for one of them, thus confirming your advice.

**Comment by Cernelius** [comment]
> You are correct in that I would have not seen the placements had I not just went in to the tool and basically deleted all of the placements from the territory affected and I noticed I had to right-click twice for one of them, thus confirming your advice.

For information (and as a suggestion), how I do it is that, whenever I use the placement picker, I write down on a piece of paper how many placement spots there are in each zone I've done. Then, once the file (txt) is generated, I open it and check that the number corresponds to what I have there, to assure I've no more than one of the same pair of coordinates (since it is impossible to see it in the tool if they are exactly the same and very hard if they differ by few pixels).

---

In my opinion, this can be solved by having the placement picker throwing a confirmation prompt each time you pick a placement that it is overlapping (by one or more pixels) with an existent one. This prompt should tell the mapmaker how many pixels are overlapping and ask him if he is sure he wants to keep this placement. Alternatively, this can happen in the moment you try to save your placement coordinates as a file (though I think this would be less usable).

**Comment by asvitkine** [comment]
On a similar topic, I noticed that Domination 1914 is really bad for the placements of units with lots of overlaps all over the map:

https://github.com/triplea-maps/domination_1914_no_mans_land/issues/9

Could someone correct that?
